app/src/main/java/com/example/screentimemanager/usageComparisonService/UsageComparisonManager.kt

```diff
@@ -0,0 +1,97 @@
+        val friendsUsageData = mutableMapOf<String, List<UsageFirebase>>()
+
+        // Fetch friends' emails
+        val friendsEmails = usageFirebaseDao.getFriendsEmails(userEmail)
+
+        // Fetch usage data for each friend
+        for (friendEmail in friendsEmails) {
+            val friendUsage = usageFirebaseDao.getUserUsageDataOnDate(friendEmail, day, month, year)
+            friendsUsageData[friendEmail] = friendUsage
+        }
+
+        return friendsUsageData
+    }
+
+
+    fun aggregateUsageData(usageData: Map<String, List<UsageFirebase>>): Map<String, Long> {
+        val aggregatedData = mutableMapOf<String, Long>()
+
+        usageData.forEach { (email, usageList) ->
+            val totalUsage = usageList.sumOf { it.usage }
+            aggregatedData[email] = totalUsage
+        }
+
+        return aggregatedData
+    }
+
+
+    suspend fun compileAndAggregateUsageData(userEmail: String, day: Int, month: Int, year: Int): Map<String, Long> {
+        val usageData = mutableMapOf<String, List<UsageFirebase>>()
+
+        // Fetch user's usage data
+        val userUsageData = fetchUserUsageData(userEmail, day, month, year)
+        usageData[userEmail] = userUsageData
+
+        // Fetch and add friends' usage data
+        val friendsUsageData = fetchFriendsUsageData(userEmail, day, month, year)
+        usageData.putAll(friendsUsageData)
+
+        // Aggregate the data
+        val aggregatedData = aggregateUsageData(usageData)
+
+        // Print the aggregated data for debugging
+        println("Aggregated Usage Data for $userEmail and friends on $day/$month/$year: $aggregatedData")
+
+        return aggregatedData
+    }
+
+
+
+    suspend fun findLowestUsageUserAndFormatMessage(userEmail: String, day: Int, month: Int, year: Int): String {
+        // Fetch and aggregate usage data
+        val aggregatedData = compileAndAggregateUsageData(userEmail, day, month, year)
+
+        // Log all the inputs
+        Log.d("UsageComparisonManager", "Notification Inputs for findLowestUsageUserAndFormat $userEmail, $day, $month, $year");
+        if (aggregatedData.isEmpty()) {
+            return "No usage data available for today."
+        }
+        val (lowestUsageUserEmail, usageTime) = aggregatedData.minByOrNull { it.value } ?: return "No usage data available."
+
+        Log.d("UsageComparisonManager", "Notification lowestUsageUserEmail: $lowestUsageUserEmail and usageTime: $usageTime");
+        // Fetch the first name of the user with the lowest usage
+        val user = userFirebaseDao.getUser(lowestUsageUserEmail)
+        val firstName = user?.firstName ?: "User"
+        
+        // Convert usage time to hours and minutes
+        val hours = usageTime / (60 * 60 * 1000)
+        val minutes = (usageTime % (60 * 60 * 1000)) / (60 * 1000)
+        val finalMessage = if (hours > 0) {
+            "$firstName got the lowest usage time today by only using $hours hours and $minutes minutes."
```

Comment:
Could you make these changes?
* add the date
* have the first name & last name
* add a case for when the user them self has the lowest screen time. In this case make the message somewhere along the line "Congrats! you had the lowest screen time among your friends on x day"
* don't show the notification to the user if they have no friends

Thanks